### PR TITLE
Dsp 19490 dse 2

### PR DESCRIPTION
**Pull request by alexliu68**

### CODE REVIEW

**Comment by jacek-lewandowski** [comment]
@jtgrabowski did we merge RM2 stuff ?

**Comment by jtgrabowski** [comment]
commit 
813e233 `Add a way to use sbt binary from configurable location` was not propagated to dse-2 and dse-2.2.

We did merge RM2 stuff, but only to current branch, which is dse-2.4.

**Comment by jtgrabowski** [comment]
@alexliu68 aren't you missing some of the commits that fixes jenkins build? Note that the commits are different for dse-2 and dse-2.2. 

https://github.com/riptano/spark-jobserver/commits/dse-2.2-build-fix
https://github.com/riptano/spark-jobserver/commits/dse-2-build-fix

**Comment by alexliu68** [comment]
@jtgrabowski My pulls are based on dse-2.2-build-fix and dse-2-build-fix.

**Comment by jtgrabowski** [comment]
Right! Apologies, everything is ok.

File: job-server/src/main/scala/spark/jobserver/JobServer.scala [hunk 1]
```diff
@@ -90,6 +97,36 @@ object JobServer {
...
+    }.mkString("\n")
+  }
+
+  private def credentialRegex(config: Config): Regex = {
+    // Use default credential keys if spark.ui.confidentialKeys is not set
+    val credentialKeys = try {
+      config.getString("spark.ui.confidentialKeys")
+    } catch {
+      case _: Exception =>
+        logger.info(s"spark.ui.confidentialKeys is not set, " +
```
Comment (jtgrabowski) [inline job-server/src/main/scala/spark/jobserver/JobServer.scala:121]: NIT: is this really an info? maybe debug would be better?

File: job-server/src/main/scala/spark/jobserver/JobServer.scala [hunk 2]
```diff
@@ -90,6 +97,36 @@ object JobServer {
...
+            Array(key, CREDENTIAL_MASK).mkString(":")
+          case _ => line
+      }
+    }.mkString("\n")
+  }
+
+  private def credentialRegex(config: Config): Regex = {
+    // Use default credential keys if spark.ui.confidentialKeys is not set
+    val credentialKeys = try {
+      config.getString("spark.ui.confidentialKeys")
```
Comment (jtgrabowski) [inline job-server/src/main/scala/spark/jobserver/JobServer.scala:118]: NIT: do we want to have this defined in config file? (to make modifications easier for users)

File: job-server/src/main/scala/spark/jobserver/JobServer.scala [hunk 3]
```diff
@@ -90,6 +97,36 @@ object JobServer {
...
+    } catch {
+      case _: Exception =>
+        logger.info(s"spark.ui.confidentialKeys is not set, " +
+          s"use default credential keys $DEFAULT_CREDENTIAL_KEYS")
+        DEFAULT_CREDENTIAL_KEYS
+    }
+
+    // Case insensitive
+    s"""(?i)(${credentialKeys.split(",").mkString("|")})" """.r
```
Comment (jtgrabowski) [inline job-server/src/main/scala/spark/jobserver/JobServer.scala:127]: why space at the end?

**Review by jtgrabowski — APPROVED** [review]
left some NITs, otherwise lgtm

File: job-server/src/main/scala/spark/jobserver/JobServer.scala (hunk 1 above)
Comment (alexliu68) [reply job-server/src/main/scala/spark/jobserver/JobServer.scala:121]: change to debug

File: job-server/src/main/scala/spark/jobserver/JobServer.scala (hunk 2 above)
Comment (alexliu68) [reply job-server/src/main/scala/spark/jobserver/JobServer.scala:118]: yes, it's from spark-defaults.conf

File: job-server/src/main/scala/spark/jobserver/JobServer.scala (hunk 3 above)
Comment (alexliu68) [reply job-server/src/main/scala/spark/jobserver/JobServer.scala:127]: There is space at the end after string split

**Pull request merged**